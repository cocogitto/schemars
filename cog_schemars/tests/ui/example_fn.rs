use cog_schemars::JsonSchema;

#[derive(JsonSchema)]
#[cog_schemars(example = "my_fn")]
pub struct Struct;

fn my_fn() {}

fn main() {}
