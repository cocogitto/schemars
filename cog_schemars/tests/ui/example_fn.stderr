error: `example` value must be an expression, and string literals that may be interpreted as function paths are currently disallowed to avoid migration errors (this restriction may be relaxed in a future version of cog_schemars).
       If you want to use the result of a function, use `#[cog_schemars(example = my_fn())]`.
       Or to use the string literal value, use `#[cog_schemars(example = &"my_fn")]`.
 --> tests/ui/example_fn.rs:4:26
  |
4 | #[cog_schemars(example = "my_fn")]
  |                          ^^^^^^^
